### PR TITLE
Migrated AlbumPreview to typescript

### PR DESCRIPTION
**Pull request by Rahul5430**

### CODE REVIEW

File: packages/ui/lib/components/AlbumPreview/index.tsx [hunk 1]
```diff
@@ -9,24 +8,58 @@ import TrackRow from '../TrackRow';
 import artPlaceholder from '../../../resources/media/art_placeholder.png';
 import styles from './styles.scss';
 
-const AlbumPreview = ({
+type Artist = {
+  name: string;
+};
+
+type Track = {
```
Comment (Gauravsharma-20) [inline packages/ui/lib/components/AlbumPreview/index.tsx]: there is `Track`  type already declared in types.ts file. use this `import { Track } from '../../types';
`

File: packages/ui/lib/components/AlbumPreview/index.tsx [hunk 2]
```diff
@@ -9,24 +8,58 @@ import TrackRow from '../TrackRow';
...
+  trackButtons?: React.ElementType ;
+};
+
+type Handlers = {
+  handleAddToQueue: () => void;
+  handlePlayAll: () => void;
+};
+
+const AlbumPreview: React.FC<AlbumPreviewProps & Handlers> = ({
   album,
```
Comment (Aimthings) [inline packages/ui/lib/components/AlbumPreview/index.tsx]: we can take out this like
`const {
 album,
  trackButtons,
  handleAddToQueue,
  handlePlayAll
}= props`

File: packages/ui/lib/components/AlbumPreview/index.tsx [hunk 3]
```diff
@@ -9,24 +8,58 @@ import TrackRow from '../TrackRow';
...
 import styles from './styles.scss';
 
-const AlbumPreview = ({
+type Artist = {
+  name: string;
+};
+
+type Track = {
+  album?: string;
+  artist?: Artist;
```
Comment (Aimthings) [inline packages/ui/lib/components/AlbumPreview/index.tsx]: ?? check this once ..object or direct string?

**Review by Aimthings — CHANGES_REQUESTED** [review]

File: packages/ui/lib/components/AlbumPreview/index.tsx (hunk 2 above)
Comment (Rahul5430) [reply packages/ui/lib/components/AlbumPreview/index.tsx]: Done
image

File: packages/ui/lib/components/AlbumPreview/index.tsx (hunk 3 above)
Comment (Rahul5430) [reply packages/ui/lib/components/AlbumPreview/index.tsx]: image
Originally its string only

File: packages/ui/lib/components/AlbumPreview/index.tsx (hunk 1 above)
Comment (Rahul5430) [reply packages/ui/lib/components/AlbumPreview/index.tsx]: done

**Review by Gauravsharma-20 — APPROVED** [review]

**Pull request merged**